### ISSUE TITLE
Inefficient regular expression complexity in EntryPoint.pattern

### ISSUE THREAD

**Issue by jaraco** [python/importlib_metadata#361]
Originally reported to the Python Security Response Team, the EntryPoint.pattern demonstrates a potential [ReDoS](https://owasp.org/www-community/attacks/Regular_expression_Denial_of_Service_).

**Comment by jaraco** [comment]
Tim Peters had this suggested fix:

```
       pattern = re.compile(
            r'(?P<module>[\w.]+)\s*'
            r'(:\s*(?P<attr>[\w.]+)\s*)?'
            r'((?P<extras>\[.*\])\s*)?$'
       )
```

**Comment by jaraco** [comment]
Released with v4.10.1 and v2.1.3 (supporting Python 2.7) (pending https://github.com/jmcgeheeiv/pyfakefs/issues/659).
